### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by SVR7PL**

### CODE REVIEW

File: api/schemas/configs/MLP_configs.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        activation: Activation function for hidden layers ('relu', 'tanh', 'sigmoid').
+        optimizer: Optimization algorithm ('sgd', 'adam', 'rmsprop').
+        learning_rate: Learning rate.
+        epochs: Number of training epochs.
+        batch_size: Batch size (None means full batch).
+        loss: Loss function ('mse', 'mae', 'cross_entropy').
+        dropout: Percentage of neurons dropped out during regularization (0-1).
+        l2_reg: L2 regularization coefficient (default None - no regularization).
+    """
+    hidden_layers: list[int] = field(default_factory=lambda: [64, 32])
```
Comment (artpods56) [inline api/schemas/configs/MLP_configs.py]: Why are we doing this like that, can't we just assign a `[64, 32` list? Please elaborate.

File: api/schemas/configs/MLP_configs.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    l2_reg: Optional[float] = None
+
+    def __post_init__(self):
+        super().__post_init__()
+        if any(h <= 0 for h in self.hidden_layers):
+            raise ValueError("All hidden layers must have a positive number of neurons.")
+        if not (0 <= self.dropout < 1):
+            raise ValueError("Dropout must be a value in the range [0, 1).")
+        if self.l2_reg is not None and self.l2_reg < 0:
+            raise ValueError("L2 regularization coefficient must be >= 0.")
```
Comment (artpods56) [inline api/schemas/configs/MLP_configs.py]: Some of the attributes are already defined in the `SupervisedAlgorithmParams` parent class. We don't need to overwrite them.

**Review by artpods56 — CHANGES_REQUESTED** [review]

File: api/schemas/configs/MLP_configs.py (hunk 1 above)
Comment (SVR7PL) [reply api/schemas/configs/MLP_configs.py:24]: The reason we use field(default_factory=...) instead of assigning the default value directly is because of the way dataclass works in Python. If we assign a list directly to hidden_layers, every instance of the MLP class will have the same list (i.e. the same reference to the same list). Consequently, if we change one instance, it will change in other instances, which leads to unintended side effects.

Therefore, field(default_factory=...) ensures that every time a new instance of MLPParams is created, the hidden_layers list will be a new, independent instance of the list [64, 32].

**Review by SVR7PL — COMMENTED** [review]
I got rid of the duplicates.

File: api/schemas/configs/MLP_configs.py (hunk 2 above)
Comment (SVR7PL) [reply api/schemas/configs/MLP_configs.py]: Done.

**Comment by agalek05** [comment]
W tym kodzie jest za mało funkcji funkcji optymalizacyjnych np.Adam.W. Jeżeli chcemy zastosować dropout to nigdy nie na poziomie 0.0, ponieważ nie dochodzi wtedy do regularyzacji.Proszę dodać dropout który będzie w zakresie 0.0-1.0, funkcję aktywacyjną softmax do klasyfikacji wieloklasowej, optmalizatory Adam.W i dodać na różnym poziomie learning-rate parametr,bo jest on kluczowy w trenowaniu sieci neuronowych. I zmienić listę 64,32 bo można w różonoraki sposób tworzyć tensory dla danych, proszę być bardziej elastyczny.

**Comment by agalek05** [comment]
In this code, there are too few optimization functions, such as AdamW. If we want to apply dropout, it should never be at a level of 0.0, as this would prevent regularization. Please add dropout within the range of 0.0 to 1.0, the softmax activation function for multi-class classification, and the AdamW optimizer. Additionally, include parameters for different learning rates at various levels, as they are crucial for training neural networks. Modify the list [64, 32] to allow more flexibility in tensor creation for the data.
